packages/gatsby-plugin-google-tagmanager/src/gatsby-ssr.js

```diff
@@ -0,0 +1,37 @@
+import React from "react"
+import { stripIndent } from "common-tags"
+
+exports.onRenderBody = (
+  { setHeadComponents, setPreBodyComponents },
+  pluginOptions
+) => {
+  if (process.env.NODE_ENV === `production`) {
+    setHeadComponents([
+      <script
+        dangerouslySetInnerHTML={{
+          __html: stripIndent`
+            <!-- Google Tag Manager -->
+            <script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+            new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+            j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+            'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+            })(window,document,'script','dataLayer', ${pluginOptions.id});</script>
+            <!-- End Google Tag Manager -->`,
+        }}
+      />,
+    ])
+
+    setPreBodyComponents([
+      <script
+        dangerouslySetInnerHTML={{
```

Comment:
Ugh I wish I'd given this thing a bit more thought. I think you're right. Does it make sense to add this if the rest of the website if not going to load without script support?

But maybe it is still useful for analytics to find out how many people land on your website without script support?

@KyleAMathews any idea?

And if I take this out, should I leave the preBodyComponents API or ditch that too?